packages/services/src/epcisEvents/transactionEvent.ts

```diff
@@ -0,0 +1,69 @@
+import { VerifiableCredential } from '@vckit/core-types';
+import { IService } from '../types/index.js';
+import { issueVC } from '../vckit.service.js';
+import { ITransactionEventContext } from './types.js';
+import { getIdentifierByObjectKeyPaths } from './helpers.js';
+import { uploadJson } from '../storage.service.js';
+import { generateUUID } from '../utils/helpers.js';
+import { registerLinkResolver } from '../linkResolver.service.js';
+import { validateTransactionEventContext } from './validateContext.js';
+import { EPCISEventAction, EPCISEventDisposition, EPCISEventType } from '../types/epcis.js';
+
+export interface ITransactionEvent {
+  data: {
+    [key: string]: any;
+  };
+}
+
+export const processTransactionEvent: IService = async (transactionEvent: ITransactionEvent, context: ITransactionEventContext): Promise<VerifiableCredential> => {
+  const validationResult = validateTransactionEventContext(context);
+  if (!validationResult.ok) {
+    throw new Error(validationResult.value);
+  }
+
+  const { vckit, epcisTransactionEvent, dlr, storage, identifierKeyPaths } = context;
+  const identifier: string = getIdentifierByObjectKeyPaths(transactionEvent.data, identifierKeyPaths);
+  if (!identifier) {
+    throw new Error('Identifier not found');
+  }
+
+  const credentialSubject = {
+    ...transactionEvent.data,
```

Comment:
Should we build a common function for reusable in other events?

Reply:
I don't think we need to separate functions for the credentialSubject because each event has very different credentialSubject data. If we build a common function, it can take a lot of parameters. So, for now, I just want to keep it simple for better readability and easier adjustment.

Reply:
Ah, I mean the fields in `credentialSubject` such as `eventID`, ` eventType`, etc. Because I see it appears many times.